src/store/modules/events/index.ts

```diff
@@ -494,27 +491,45 @@ const module: Module<EventsModuleState, RootState> = {
       const skip = (state.repetitions[key] || []).length;
 
       const response = await eventsApi.getLatestRepetitions(projectId, eventId, skip, limit);
-      const repetitions = response.data.project.event.repetitions;
+
+      if (originalEvent) {
+        filterBeautifiedAddons([originalEvent]);
+      }
 
       /**
-       * Solution for not displaying both `userAgent` and `beautifiedUserAgent` addons
+       * If delta is present, apply delta to the event payload
        */
-      filterBeautifiedAddons(repetitions);
+      const repetitions = response.data.project.event.repetitions.map(repetition => {
+        let composedRepetition: HawkEventRepetition;
```

Comment:
```suggestion
        let composedRepetition: HawkEventRepetition = repetition;
```
then you can remove one of the conditions below